File: Python/assignment ii/9.py
def wordPrint(num):
    numWord = { 'tens': [],
               'hundreds': [],
               'thousands': [] }
    
    if num % 100 < 20:
        if num == 1:
            numWord['tens'].append('one')
        elif num == 2:
            numWord['tens'].append('two')
        elif num == 3:
            numWord['tens'].append('three')
        elif num == 4:
            numWord['tens'].append('four')
        elif num == 5:
            numWord['tens'].append('five')
        elif num == 6:
            numWord['tens'].append('six')
        elif num == 7:
            numWord['tens'].append('seven')
        elif num == 8:
            numWord['tens'].append('eight')
        elif num == 9:
            numWord['tens'].append('nine')
        elif num == 10:
            numWord['tens'].append('ten')
        elif num == 11:
            numWord['tens'].append('eleven')
        elif num == 12:
            numWord['tens'].append('twelve')
        elif num == 13:
            numWord['tens'].append('thirteen')
        elif num == 14:
            numWord['tens'].append('fourteen')
        elif num == 15:
            numWord['tens'].append('fifteen')
        elif num == 16:
            numWord['tens'].append('sixteen')
        elif num == 17:
            numWord['tens'].append('seventeen')
        elif num == 18:
            numWord['tens'].append('eighteen')
        elif num == 19:
            numWord['tens'].append('nineteen')
    else:
        if (num % 100) - (num % 10) == 20:
            numWord['tens'].append('twenty')
        elif (num % 100) - (num % 10) == 30:
            numWord['tens'].append('thirty')
        elif (num % 100) - (num % 10) == 40:
            numWord['tens'].append('forty')
        elif (num % 100) - (num % 10) == 50:
            numWord['tens'].append('fifty')
        elif (num % 100) - (num % 10) == 60:
            numWord['tens'].append('sixty')
        elif (num % 100) - (num % 10) == 70:
            numWord['tens'].append('seventy')
        elif (num % 100) - (num % 10) == 80:
            numWord['tens'].append('eighty')
        elif (num % 100) - (num % 10) == 90:
            numWord['tens'].append('ninety')
            
        if num % 10 == 1:
            numWord['tens'].append('one')
        elif num % 10 == 2:
            numWord['tens'].append('two')
        elif num % 10 == 3:
            numWord['tens'].append('three')
        elif num % 10 == 4:
            numWord['tens'].append('four')
        elif num % 10 == 5:
            numWord['tens'].append('five')
        elif num % 10 == 6:
            numWord['tens'].append('six')
        elif num % 10 == 7:
            numWord['tens'].append('seven')
        elif num % 10 == 8:
            numWord['tens'].append('eight')
        elif num % 10 == 9:
            numWord['tens'].append('nine')
            
    if (num % 1000) - (num % 100) == 100:
        numWord['hundreds'].append('one')
    elif (num % 1000) - (num % 100) == 200:
        numWord['hundreds'].append('two')
    elif (num % 1000) - (num % 100) == 300:
        numWord['hundreds'].append('three')
    elif (num % 1000) - (num % 100) == 400:
        numWord['hundreds'].append('four')
    elif (num % 1000) - (num % 100) == 500:
        numWord['hundreds'].append('five')
    elif (num % 1000) - (num % 100) == 600:
        numWord['hundreds'].append('six')
    elif (num % 1000) - (num % 100) == 700:
        numWord['hundreds'].append('seven')
    elif (num % 1000) - (num % 100) == 800:
        numWord['hundreds'].append('eight')
    elif (num % 1000) - (num % 100) == 900:
        numWord['hundreds'].append('nine')
    
    if (num % 1000) - (num % 100) != 0:
        numWord['hundreds'].append('hundred')
    
    if num - (num % 1000) == 1000:
        numWord['thousands'].append('one')
    elif num - (num % 1000) == 2000:
        numWord['thousands'].append('two')
    elif num - (num % 1000) == 3000:
        numWord['thousands'].append('three')
    elif num - (num % 1000) == 4000:
        numWord['thousands'].append('four')
    elif num - (num % 1000) == 5000:
        numWord['thousands'].append('five')
    elif num - (num % 1000) == 6000:
        numWord['thousands'].append('six')
    elif num - (num % 1000) == 7000:
        numWord['thousands'].append('seven')
    elif num - (num % 1000) == 8000:
        numWord['thousands'].append('eight')
    elif num - (num % 1000) == 9000:
        numWord['thousands'].append('nine')
        
    if num - (num % 1000) != 0:
        numWord['thousands'].append('thousand')
        
    print(numWord['thousands'][0] + ' ' + numWord['thousands'][1] + ' ' + numWord['hundreds'][0] + ' ' + numWord['hundreds'][1] + ' ' + numWord['tens'][0] + ' ' + numWord['tens'][1])

num = int(input("Enter the number: "))

wordPrint(num)

File: C/SEM I/Assignment II/9.c
/* 

Assignemnt II
Name: Adersh Varshnei
9. Write a C Program to enter a number and print it in words

*/

#include <stdio.h>
#include <conio.h>
#include <stdlib.h>
#include <math.h>

//************************************************************************************************

typedef unsigned int loopVar;

//************************************************************************************************

int * numToArr(int);
void printDigitWord(int);
void numToWord(int);

//************************************************************************************************

int count = 0;

//************************************************************************************************

int main(void) {
	loopVar rep = 1;

	while (rep == 1) {
		int inputNum;

		do {
			printf("Enter the number(Between -9999 and 9999): ");
			scanf("%d", & inputNum);

			if (abs(inputNum) > 10000) {
				printf("Invalid entry. Please enter number again.\n");
			} //if (abs(inputNum) > 10000)
		} while (abs(inputNum) > 10000);

		printf("The number in words is: ");
		numToWord(inputNum);

		printf("\nRepeat Program?: ");
		scanf("%d", & rep);

		count++;

		if (count >= 15) {
			printf("Abnormal END\n");

			getch();
			exit(0);
		} //if (count >= 15) 
	} //while (rep == 1) 

	printf("Normal END\n");

	getch();
	return 0;
} //int main(void)

//************************************************************************************************

int * numToArr(int num) {
	static int numArr[5];
	loopVar i;
	static int repChk;

	if (count > repChk) {
		for (i = 0; i < 5; i++) {
			numArr[i] = 0;
		} //for (i = 0; i < 5; i++)

		repChk = count;
	} //if (count > repChk)

	if (num < 0) {
		numArr[0] = 1;
	} else {
		numArr[0] = 0;
	} //End of extended if clause

	num = abs(num);

	for (i = 4; num > 0; num /= 10, i--) {
		numArr[i] = num % 10;
	} //for (i = 4; num > 0; num /= 10, i--)

	return numArr;
} //int * numToArr(int num)

//************************************************************************************************

void printDigitWord(int num) {
	switch (num) {
		case 0:
			break;

		case 1:
			printf("One");
			break;

		case 2:
			printf("Two");
			break;

		case 3:
			printf("Three");
			break;

		case 4:
			printf("Four");
			break;

		case 5:
			printf("Five");
			break;

		case 6:
			printf("Six");
			break;

		case 7:
			printf("Seven");
			break;

		case 8:
			printf("Eight");
			break;

		case 9:
			printf("Nine");
			break;
	} //switch (num)
} //void printDigitWord(int num)

//************************************************************************************************

void numToWord(int num) {
	int * numArr;
	loopVar i;

	if (num == 0) {
		printf("Zero");
	} else {
		numArr = numToArr(num);

		if (numArr[0] == 1) {
			printf("Negative ");
		} //if (numArr[0] == 1)

		printDigitWord(numArr[1]);

		if (numArr[1] != 0) {
			printf(" Thousand ");
		} //if (numArr[1] != 0)

		printDigitWord(numArr[2]);

		if (numArr[2] != 0) {
			printf(" Hundred ");
		} //if (numArr[2] != 0)

		if (numArr[3] * 10 + numArr[4] < 20) {
			switch (numArr[3] * 10 + numArr[4]) {
				case 0:
					printf("");
					break;

				case 1:
					printf("One");
					break;

				case 2:
					printf("Two");
					break;

				case 3:
					printf("Three");
					break;

				case 4:
					printf("Four");
					break;

				case 5:
					printf("Five");
					break;

				case 6:
					printf("Six");
					break;

				case 7:
					printf("Seven");
					break;

				case 8:
					printf("Eight");
					break;

				case 9:
					printf("Nine");
					break;

				case 10:
					printf("Ten");
					break;

				case 11:
					printf("Eleven");
					break;

				case 12:
					printf("Twelve");
					break;

				case 13:
					printf("Thirteen");
					break;

				case 14:
					printf("Fourteen");
					break;

				case 15:
					printf("Fifteen");
					break;

				case 16:
					printf("Sixteen");
					break;

				case 17:
					printf("Seventeen");
					break;

				case 18:
					printf("Eighteen");
					break;

				case 19:
					printf("Nineteen");
					break;
			} //switch (numArr[3] * 10 + numArr[4])
		} else {
			switch (numArr[3]) {
				case 2:
					printf("Twenty");
					break;

				case 3:
					printf("Thirty");
					break;

				case 4:
					printf("Fourty");
					break;

				case 5:
					printf("Fifty");
					break;

				case 6:
					printf("Sixty");
					break;

				case 7:
					printf("Seventy");
					break;

				case 8:
					printf("Eighty");
					break;

				case 9:
					printf("Ninety");
					break;
			} //switch (numArr[3])

			if(numArr[4]>0){
				printf("-");
			}

			printDigitWord(numArr[4]);
		} //End of extended if clause
	} //End of extended if clause
} //void numToWord(int num)